File: Java/Program/Shirt.java
public class Shirt extends Clothing {
    private String color; // Atribut color anggota DPR
    private String sleeveType; // Atribut nama anggota DPR

    // Atribut statis untuk menyimpan panjang maksimum dari setiap atribut
    private static int maxcolor;
    private static int maxsleeveType;

    /**
     * Konstruktor default untuk objek Product.
     * Inisialisasi atribut color, sleeveType, , dan dengan string kosong.
     */
    public Shirt() {
        super();
        this.color = "";
        this.sleeveType = "";
    }

    /**
     * Konstruktor untuk objek Product dengan parameter color, sleeveType, ,
     * dan
     * .
     * Menginisialisasi atribut objek dan memperbarui nilai maxcolor, maxsleeveType,
     * max, dan max
     * jika panjang atribut yang diinisialisasi melebihi nilai maksimum sebelumnya.
     */
    public Shirt(String id, String name, String brand, String price, String size, String material, String gender, String color, String sleeveType) {
        super(id, name, brand, price, size, material, gender);
        this.color = color;
        this.sleeveType = sleeveType;

        // Memperbarui nilai maksimum jika panjang atribut melebihi nilai maksimum
        // sebelumnya
        if (this.color.length() > maxcolor) {
            maxcolor = this.color.length();
        }

        if (this.sleeveType.length() > maxsleeveType) {
            maxsleeveType = this.sleeveType.length();
        }
    }

    /* Getter & Setter */

    /**
     * Getter untuk atribut color.
     * 
     * @return Nilai atribut color.
     */
    public String getcolor() {
        return this.color;
    }

    /**
     * Setter untuk atribut color.
     * 
     * @param color Nilai yang akan di-set pada atribut color.
     */
    public void setcolor(String color) {
        this.color = color;
    }

    /**
     * Getter untuk atribut sleeveType.
     * 
     * @return Nilai atribut sleeveType.
     */
    public String getsleeveType() {
        return this.sleeveType;
    }

    /**
     * Setter untuk atribut sleeveType.
     * 
     * @param sleeveType Nilai yang akan di-set pada atribut sleeveType.
     */
    public void setsleeveType(String sleeveType) {
        this.sleeveType = sleeveType;
    }

    /**
     * Getter untuk nilai maksimum panjang atribut color.
     * 
     * @return Nilai maksimum panjang atribut color.
     */
    public static int getMaxcolor() {
        return maxcolor;
    }

    /**
     * Getter untuk nilai maksimum panjang atribut sleeveType.
     * 
     * @return Nilai maksimum panjang atribut sleeveType.
     */
    public static int getMaxsleeveType() {
        return maxsleeveType;
    }
}

File: CPP/Program/Shirt.cpp
#include <iostream>
#include <string>
#include "Clothing.cpp"

using namespace std;

// Deklarasi kelas Shirt yang merupakan turunan dari kelas Clothing
class Shirt : public Clothing {
private:
    string color;
    string sleeveType;

    // Atribut statis untuk menyimpan panjang maksimum dari setiap atribut
    static int maxColor;
    static int maxSleeveType;

public:
    // Konstruktor default Shirt yang memanggil konstruktor default dari kelas Clothing
    Shirt() : Clothing() {
        color = "";
        sleeveType = "";
    }

    // Konstruktor Shirt dengan parameter yang memanggil konstruktor parameter dari kelas Clothing
    Shirt(string id, string name, string brand, string price, string size, string material, string gender, string color, string sleeveType)
        : Clothing(id, name, brand, price, size, material, gender), color(color), sleeveType(sleeveType) {

        // Memperbarui nilai maksimum jika panjang atribut melebihi nilai maksimum sebelumnya
        if (this->color.length() > maxColor) {
            maxColor = this->color.length();
        }

        if (this->sleeveType.length() > maxSleeveType) {
            maxSleeveType = this->sleeveType.length();
        }
    }

    // Getter dan Setter untuk atribut color
    string get_color() {
        return color;
    }

    void set_color(string color) {
        this->color = color;
    }

    // Getter dan Setter untuk atribut sleeveType
    string get_sleeveType() {
        return sleeveType;
    }

    void set_sleeveType(string sleeveType) {
        this->sleeveType = sleeveType;
    }

    // Getter untuk nilai maksimum panjang atribut color
    static int getmaxcolor() {
        return maxColor;
    }

    // Getter untuk nilai maksimum panjang atribut sleeveType
    static int getmaxsleeveType() {
        return maxSleeveType;
    }
};
